### ISSUE TITLE
Support loading whole package components

### ISSUE THREAD

**Issue by mgsloan** [commercialhaskell/intero#39]
It'd be great to support loading whole components / packages.  For example, you could do `:load stack` to load the entire stack package.  There are a number of advantages of first-class support for using info from the build system:

1) No need to restart ghci when cabal files change.  There may be some cases where a restart is needed for some flags (see https://github.com/fpco/ide-backend/issues/115).  Also not sure how well this will work when the stack.yaml changes and causes a different set of dependencies.

2) Easily specify which part of the project you want to load, without doing a fresh invocation of `stack ghci`

3) Possible to have a much more concise prompt.  Currently `stack ghci` on big projects is nearly unusable without using `:set prompt ...`, because of the huge prompts.  Instead we can have it simply be something like `stack>`

**Comment by mgsloan** [comment]
Once https://github.com/commercialhaskell/stack/issues/1361 is implemented, it should also be imminently feasible to have `:load ModName` and `:load src/ModName.hs` also bring in the requisite flags.  This will likely depend on #41

**Comment by chrisdone** [comment]
Right, loading a target sounds cool. If we can get the necessary info. Tho
in the short term it seems like ghc isn't good at reloading things
dynamically despite the name. A worthy goal tho.

**Comment by k-bx** [comment]
@mgsloan  I'm not sure if this is the same problem I'm experiencing on my work-project or this needs a separate issue created, please comment. Our project is split into multiple projects all inside a single big repo with a single stack.yaml for all subdirs with projects, so whenever I edit dependant project, intero starts breaking in different ways, and I only am able to fix it by running "stack build" outside and then restarting intero. That includes edits of both, hs-files and cabal-file.

**Comment by chrisdone** [comment]
@k-bx By default Intero boots up for the closest .cabal package. That probably means the other packages are loaded as packages, not as source files. If you want to load them all in at once, and they're compatible in terms of options and such, you could do `M-x intero-targets` and specify `thisone thatone` and it'll try to load both packages at once into one GHCi.

**Comment by chrisdone** [comment]
Reducing Intero's scope, closing.
